airflow/www/decorators.py

```diff
@@ -34,6 +34,15 @@
 logger = logging.getLogger(__name__)
 
 
+def _get_dag_id() -> str:
+    dag_id = request.args.get('dag_id') 
+    if dag_id:
+        return dag_id
+    path_parts = request.path.split('/')
+    if len(path_parts) >= 3 and path_parts[1] == 'dags':
+        return path_parts[2]
```

Comment:
Instead of relying on the `/dags/<dag_id>` structure, it is likely better to use [`request.endpoint`](https://flask.palletsprojects.com/en/1.1.x/api/?highlight=request#flask.Request.endpoint) to get the `dag_id` component in the path instead.

Reply:
Thanks! I was looking for it in a wrong version of Flask, but failed to locate `request.view_args` there for some reason. Fixed.